resources/views/payment.blade.php

```diff
@@ -33,27 +33,27 @@
                 <span class="ml-3">@{{ successMessage }}</span>
             </p>
 
-            <div class="bg-white rounded-lg shadow-xl p-4 sm:py-6 sm:px-10 mb-5">
+            <div class="bg-white rounded-lg shadow-xl p-4 sm:p-6 mt-4">
                 @if ($payment->isSucceeded())
-                    <h1 class="text-xl mt-2 mb-4 text-gray-700">
+                    <h1 class="text-xl mb-4 text-gray-600">
                         {{ __('Payment Successful') }}
                     </h1>
 
                     <p class="mb-6">
                         {{ __('This payment was already successfully confirmed.') }}
                     </p>
                 @elseif ($payment->isCancelled())
-                    <h1 class="text-xl mt-2 mb-4 text-gray-700">
+                    <h1 class="text-xl mb-4 text-gray-600">
                         {{ __('Payment Cancelled') }}
                     </h1>
 
                     <p class="mb-6">{{ __('This payment was cancelled.') }}</p>
                 @else
                     <div id="payment-elements" v-if="! paymentProcessed">
                         <!-- Payment Method Form -->
-                        <div v-show="requiresPaymentMethod">
+                        <div v-show="requiresPaymentMethod || requiresConfirmation">
```

Comment:
I think this is the wrong place for this check, it's much more like `requiresAction` as it doesn't require the user to enter payment method details.

https://github.com/laravel/cashier-stripe/blob/b21e2d625d9fa83b5fe847b9567420c466b20620/resources/views/payment.blade.php#L97

Reply:
See also:

https://stripe.com/docs/payments/intents